documentation/high-level-introduction.md

```diff
@@ -0,0 +1,257 @@
+# A brief introduction to msbuild (Level 101)
+
+## Introduction
+
+This document aims to be a “101 college” level introduction into msbuild, useful for those coming to msbuild with very little background in how it works.
+
+### What is msbuild?
+
+msbuild is a build system commonly used to build .NET projects. It is made up the msbuild toolchain itself, and some number of additional files provided by the SDKs which provide platform specific instructions on how to process projects of those types.
+
+msbuild is built up of a few very generic ideas:
+
+- Run a sequence of defined steps based upon input / output files and dependency ordering
+- Store single and groups of values in a global table
+- Steps can load C# library tasks and execute them
+- Include other files in the file we are processing 
+
+Built on top of those bare bones are a huge number of conventions:
+
+- Separating your data and step definitions in different files (see file structure below)
+- Creating common locations to import “SDK” files from
+- Standard variables to add source files \ references \ resources
```

Comment:
Backslashes instead of slashes are used throughout this file. Change them to forward slashes?